x-pack/legacy/plugins/apm/public/components/app/ServiceMap/useRefHeight.ts

```diff
@@ -8,13 +8,17 @@ import { useWindowSize } from 'react-use';
 
 export function useRefHeight(): [
   MutableRefObject<HTMLDivElement | null>,
+  number,
   number
 ] {
   const ref = useRef<HTMLDivElement>(null);
   const windowHeight = useWindowSize().height;
-  const topOffset = ref.current?.getBoundingClientRect()?.top ?? 0;
+  const { top: topOffset, width } = ref.current?.getBoundingClientRect() ?? {
+    top: 0,
+    width: 0
+  };
 
   const height = ref.current ? windowHeight - topOffset : 0;
 
-  return [ref, height];
+  return [ref, height, width];
 }
```

Comment:
A quick scan: looks like it would be more clear to return the default value early and the conditionals disappear (untested):

```tsx
export function useRefHeight(): [
  MutableRefObject<HTMLDivElement | null>,
  number,
  number
] {
  const ref = useRef<HTMLDivElement>(null);

  if (!ref.current) {
    return [ref, 0, 0];
  }

  const windowHeight = useWindowSize().height;
  const { top, width } = ref.current.getBoundingClientRect();
  const height = windowHeight - top;

  return [ref, height, width];
}
```

Btw what do you think about returning an object similar to how `useWindowSize` works?
```
return { ref, height, width };
```